src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -283,6 +293,46 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// Completion extension load request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleCompletionExtLoadRequest(CompletionExtensionParams param, RequestContext<bool> requestContext)
+        {
+            try
+            {
+                var serviceProvider = ExtensionServiceProvider.Create(new Assembly[] { AssemblyLoadContext.Default.LoadFromAssemblyPath(param.Assembly) });
+                foreach (var provider in serviceProvider.GetServices<ICompletionExtensionProvider>())
+                {
+                    var cancellationToken = new CancellationTokenSource(ExtensionLoadingTimeout).Token;
+                    var ext = await provider.CreateAsync(param.Properties ?? new Dictionary<string, object>(), cancellationToken);
+                    if (ext == null)
```

Comment:
Hm, we shouldn't just be relying on just the extension to properly cancel the request. There's nothing preventing it from just ignoring the cancellation and taking a huge amount of time. Would prefer we wrap these in our own logic to handle the timeout instead. Same for below with the completion request

Reply:
Added timeout in the calling code.